### PR TITLE
Added a JMS notification receiver and examples for os messages.

### PR DESCRIPTION
**Pull request by andy-maier**
Please review.

Testcases are not yet part of this PR, but I wanted to get the API of the `NotificationReceiver` class reviewed as soon as possible.

Details:
- Added a NotificationReceiver class that is a general purpose JMS message / notification receiver.
- Added a chapter 'Notifications' to the documentation, describing the new class.
- Added an example show_os_messages.py that uses the new open_os_message_channel() method added in a previous commit, and the new notification receiver, to display the OS messages of a partition or LPAR.

### CODE REVIEW

File: docs/notifications.rst [hunk 1]
```diff
@@ -0,0 +1,25 @@
+.. Copyright 2016 IBM Corp. All Rights Reserved.
```
Comment (leopoldjuergen) [inline docs/notifications.rst]: Could you change the copyright to 2017 please ?

File: examples/show_os_messages.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python
+# Copyright 2016 IBM Corp. All Rights Reserved.
```
Comment (leopoldjuergen) [inline examples/show_os_messages.py]: Please change to 2017

Comment (andy-maier) [inline examples/show_os_messages.py]: Done

File: docs/notifications.rst (hunk 1 above)
Comment (andy-maier) [inline docs/notifications.rst]: Done

**Comment by leopoldjuergen** [comment]
+1

**Pull request merged**